### PR TITLE
cw pairs of shoes

### PR DESCRIPTION
**Pull request by aepaysinger**
what i want shoes to look like
shoes = {
21: {0, 1},
23: {1, 0}
}
the key should be the shoe size and the value should be a set of 1 and 0. 
if i change the code to shoes[size] = shoes.get(size, foot), foot
(which i thought would add the value if not there) it does {21: ((0, 0), 1), 23: ((1, 1), 0)}
which is what i am looking.
any advice on what i can research to help???

### CODE REVIEW

**Comment by W-Ely** [comment]
`shoes[size] = shoes.get(size, foot), foot` ---> `shoes.get(size, foot)` evaluates to what is already the value in the dict or `foot` .  `shoes.get(size, foot), foot` is a tuple `(shoes.get(size, foot), foot)`

```python
from collections import defaultdict

def make_pairs(mixed_shoes):
    shoes = defaultdict(set)
    for foot, size in mixed_shoes:
        shoes[size].add(foot)
    print(shoes)
 ```

**Comment by W-Ely** [comment]
or if you don't want to use the builtin,
```python
def make_pairs(mixed_shoes):
    shoes = {}
    for foot, size in mixed_shoes:
        shoes[size] = shoes.get(size, set()) | set((foot,))
    print(shoes)
```

**Comment by aepaysinger** [comment]
set()) | set((foot,)
so this is how the set is updating..... | (union) returns the updates set?

File: code_challenges/code_wars/pairs_of_shoes.py [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
-    mixed_shoes = [[0, 21], 
-         [1, 23], 
-         [1, 21], 
-         [0, 23]]
-    print(make_pairs(mixed_shoes))
\ No newline at end of file
+    mixed_shoes = [[0, 21], [1, 23], [1, 21], [1, 23]]
+    print(make_pairs(mixed_shoes))
+
+    # use a counter --> collections counter
```
Comment (W-Ely) [inline code_challenges/code_wars/pairs_of_shoes.py]: The thought was to read collections.Counter and see if there was a way to use it, might or might not be applicable.

File: code_challenges/code_wars/pairs_of_shoes.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+def find_pairs(mixed_shoes):
+    shoes = make_pairs(mixed_shoes)
+    shoe_pairs = False
+    for size in shoes:
+        if shoes[size][0] == shoes[size][1]:
+            shoe_pairs = True
+        else:
+            return False
+    return shoe_pairs
```
Comment (W-Ely) [inline code_challenges/code_wars/pairs_of_shoes.py:22]:
```suggestion
    for size in shoes:
        if shoes[size][0] != shoes[size][1]:
            return False
    return True
```

File: code_challenges/code_wars/pairs_of_shoes.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+def make_pairs(mixed_shoes):
+    shoes = {}
+
+    for foot, size in mixed_shoes:
+        if size not in shoes:
+            shoes[size] = {0: 0, 1: 0}
+            shoes[size][foot] += 1
+        else:
+            shoes[size][foot] += 1
```
Comment (W-Ely) [inline code_challenges/code_wars/pairs_of_shoes.py:9]: Nice!

**Review by W-Ely — APPROVED** [review]

**Pull request merged**